### PR TITLE
add usdm rule runs in test suite actions

### PR DESCRIPTION
**Pull request by RamilCDISC**

### CODE REVIEW

**Comment by RamilCDISC** [comment]
The PR for adding the test datasest and rule lists to test suite repository is here: https://github.com/cdisc-org/CORE_Test_Suite/pull/40

File: .github/workflows/test_suite.yml [hunk 1]
```diff
@@ -85,27 +97,83 @@ jobs:
...
+
+          if [ -f "CORE_Test_Suite/usdm_positive_report.json" ]; then
+            echo "usdm_pos_success=true" >> $GITHUB_OUTPUT
+            echo "## USDM Positive" >> $GITHUB_STEP_SUMMARY
+            echo "**Success**: Positive test passed" >> $GITHUB_STEP_SUMMARY
+            python CORE_Test_Suite/scripts/validation_summary.py CORE_Test_Suite/usdm_positive_report.json >> $GITHUB_STEP_SUMMARY
+          else
+            echo "usdm_pos_success=false" >> $GITHUB_OUTPUT
+            echo "**Failed**: No results for positive test" >> $GITHUB_STEP_SUMMARY
+          fi
```
Comment (SFJohnson24) [inline .github/workflows/test_suite.yml:160]: https://github.com/cdisc-org/cdisc-rules-engine/actions/runs/14841150152/job/41664189554?pr=1157
https://github.com/cdisc-org/cdisc-rules-engine/actions/runs/14841430308/job/41665097807
 This is not working currently.  Please see the logs.  It is not finding the negative report or the positive dataset

File: .github/workflows/test_suite.yml [hunk 2]
```diff
@@ -119,7 +187,17 @@ jobs:
...
+            CORE_Test_Suite/usdm_negative_comparison.json
+            CORE_Test_Suite/usdm_positive_comparison.json
           if-no-files-found: warn
+
+      ###########################
+      # FINAL DIFF CHECK RESULT
+      ###########################
       - name: Check for differences
         if: always()
         run: |
```
Comment (SFJohnson24) [inline .github/workflows/test_suite.yml:203]: you need add 2 USDM Diff conditions to the environment so that the test suite will fail/succeed based on the usdm negative and positive failing or succeeding

**Review by SFJohnson24 — CHANGES_REQUESTED** [review]

File: .github/workflows/test_suite.yml [hunk 3]
```diff
@@ -85,27 +97,83 @@ jobs:
...
+            echo "**Failed**: No results for positive test" >> $GITHUB_STEP_SUMMARY
+          fi
+
+      - name: Compare USDM negative result
+        if: steps.usdm_neg.outputs.usdm_neg_success == 'true'
+        continue-on-error: true
+        run: |
+          python CORE_Test_Suite/scripts/comparison.py CORE_Test_Suite/usdm_negative_report.json CORE_Test_Suite/USDM_Negative_Result.json CORE_Test_Suite/usdm_negative_comparison.xlsx --mode test --json-output CORE_Test_Suite/usdm_negative_comparison.json
+
+      - name: Compare USDM positive result
```
Comment (SFJohnson24) [inline .github/workflows/test_suite.yml:168]: can you move the comparison up to right after the validation runs please for continuity.

**Review by SFJohnson24 — CHANGES_REQUESTED** [review]
https://github.com/cdisc-org/cdisc-rules-engine/actions/runs/14911092214  This all looks good and seems to be functioning as expected.  Just a slight change in the step sequencing and we are good to merge

File: .github/workflows/test_suite.yml (hunk 3 above)
Comment (RamilCDISC) [reply .github/workflows/test_suite.yml:168]: updated

**Review by SFJohnson24 — APPROVED** [review]
looks good
